pkg/controller/populators/upload-populator.go

```diff
@@ -131,12 +131,14 @@ func (r *UploadPopulatorReconciler) updatePVCPrimeNameAnnotation(pvc *corev1.Per
 	return true, nil
 }
 
-func removePVCPrimeNameAnnotation(pvc *corev1.PersistentVolumeClaim) {
+func (r *UploadPopulatorReconciler) updateUploadAnnotations(pvc *corev1.PersistentVolumeClaim, pvcPrime *corev1.PersistentVolumeClaim) {
 	if _, ok := pvc.Annotations[AnnPVCPrimeName]; !ok {
 		return
 	}
-
-	delete(pvc.Annotations, AnnPVCPrimeName)
+	// Delete the PVC Prime annotation once the pod is succeeded
+	if pvcPrime.Annotations[cc.AnnPodPhase] == string(corev1.PodSucceeded) {
+		delete(pvc.Annotations, AnnPVCPrimeName)
```

Comment:
perhaps a unit test for this?

Reply:
Okay. This flow was already tested but just added a new one to specifically cover `r.updatePVCWithPVCPrimeAnnotations(pvcCopy, pvcPrime, r.updateUploadAnnotations)`. Still having lots of upload errors in the tests though, I'm clueless about what's failing.